### DISCUSSION TITLE
Order in config with --config

### DISCUSSION THREAD

**Discussion by MoeLa** [artilleryio/artillery#948] [Artillery]
I have a _config.yaml_ like
```
config:
  variables:
    repeat: "5"
```
and a _test-script.yaml_ starts with
```
config:
  variables:
    repeat: "2"
```
When I start this with `artillery run --config config.yaml test-script.yaml`, which value will `{{ repeat }}` have?

**Comment by hassy** [comment]
external config takes precedence so it should be `5`

**Reply by MoeLa** [reply]
Ok, thanks.

In my situation, I'd need it the other way around: I have defined the default value (with lots of other stuff) in _config.yaml_ that I usually want to use, but in one special test, I'd like so use a differing value for `{{ repeat }}`.

**Reply by milosgajdos** [reply]
I have a similar need, actually. I think rather than passing a JSON via `--variables` it'd be great if you could define `variables.yaml` or some such and pass those in. Helm chart work sort of that way: you define `values.yaml` and then do `helm install --values values.yaml` and whatever is in `values.yaml` overrides what's in your other template. Passing JSON via cli seems like a very odd choice 🤔
